lib/util.js

```diff
@@ -352,6 +352,12 @@ exports.stripHTML = html => html
   .replace(/[\n\r]/g, ' ')
   .replace(/\s*<\s*br\s*\/?\s*>\s*/gi, '\n')
   .replace(/<\s*\/\s*p\s*>\s*<\s*p[^>]*>/gi, '\n')
+  .replace(/<a\s+(?:[^>]*?\s+)?href=(?:["'])\/redirect.*?q=(.*?)(?:[&'"]).*?<\/a>/gi,
+    (_, p1) => decodeURIComponent(p1) || p1)
+  .replace(/<a\s+(?:[^>]*?\s+)?href=(?:["'])(https?.*?)(?:['"]).*?<\/a>/gi,
+    (_, p1) => decodeURIComponent(p1) || p1)
+  .replace(/<a\s+(?:[^>]*?\s+)?href=(?:["'])((?:\/results?|\/watch?).*?)(?:['"]).*?<\/a>/gi,
+    (_, p1) => `https://youtube.com${decodeURIComponent(p1)}` || p1)
```

Comment:
you can combine these two and use `url.resolve('https://youtube.com/', p1)`. `resolve()` will return the original url if the domain doesn't match youtube.com, and it will futureproof other relative paths within youtube.